tutorials/stats-sensor-space/76_new_cluster_test_api.py

```diff
@@ -0,0 +1,491 @@
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+import mne
+
+# TODO: implement formulaic design matrix for paired t-test
+# TODO: @erik: add dataset to mne-data
+
+# import and load dataset
+path_to_p3 = Path("C:/Users/Carina/mne_data/ERP_CORE_P3")
+
+
+def prep_sample_data(plot_evokeds: bool = False):
+    """
+    Load the P3 dataset and extract the target, non-target and contrast evokeds.
+    """
+    # Define the range of participant IDs
+    participant_ids = range(15, 20)  # This will cover 015 to 019
+
+    evokeds_allsubs = []
+
+    # Loop over each participant ID and generate the corresponding filename
+    for pid in participant_ids:
+        # Create the filename using an f-string, ensuring the participant ID is zero-padded to 3 digits
+        filename_p3 = f"sub-{pid:03d}_ses-P3_task-P3_ave.fif"
+
+        # Print the filename (or perform your desired operations on it)
+        print(filename_p3)
+
+        p3_file_path = Path(path_to_p3) / filename_p3
+
+        evokeds = mne.read_evokeds(p3_file_path)
+
+        # add to list
+        evokeds_allsubs.append(evokeds)
+
+    target_only = [evoked[0] for evoked in evokeds_allsubs]
+    non_target_only = [evoked[1] for evoked in evokeds_allsubs]
+    contrast = [evoked[2] for evoked in evokeds_allsubs]
+
+    if plot_evokeds:
+        # plot the grand average
+        mne.grand_average(target_only).plot()
+        mne.grand_average(non_target_only).plot()
+        mne.grand_average(contrast).plot()
+
+    # create contrast from evokeds target and non-target
+    diff_evoked = [
+        mne.combine_evoked([evokeds_a, evokeds_b], weights=[1, -1])
+        for evokeds_a, evokeds_b in zip(target_only, non_target_only)
+    ]
+
+    if plot_evokeds:
+        mne.grand_average(diff_evoked).plot()
+
+    # crop the evokeds in the post stimulus window
+    contrast = [evokeds.crop(tmin=-0.1, tmax=0.6) for evokeds in contrast]
+    target_only = [evokeds.crop(tmin=-0.1, tmax=0.6) for evokeds in target_only]
+    non_target_only = [evokeds.crop(tmin=-0.1, tmax=0.6) for evokeds in non_target_only]
+
+    return contrast, target_only, non_target_only
+
+
+def old_api_cluster(n_permutations: int = 10000, seed: int = 1234):
```

Comment:
eventually this will become part of a test, rather than the tutorial, I think

Reply:
Agreed